src/checkout/useCheckIsFree.js

```diff
@@ -0,0 +1,18 @@
+import { useContext } from 'react';
+import { CheckoutContext } from './CheckoutProvider';
+import useCommerce from '../useCommerce';
+
+export default function useCheckIsFree() {
+  const commerce = useCommerce();
+  const { checkout } = useContext(CheckoutContext);
+
+  if (!checkout || !commerce) {
+    return null;
+  }
+
+  return commerce.checkout.isFree(
+    checkout.id,
+  ).then((result) => {
+    return result;
+  });
```

Comment:
```suggestion
  useCallback(async () => {
    if (!checkout || !commerce) {
      return null;
    }

    return commerce.checkout.isFree(
      checkout.id,
    ).then((result) => {
      return result;
    });
  }, [checkout, commerce]);
```

BYO import for `useCallback`.

Reply:
ok gotcha thanks!

Reply:
The `.then` here is also pointless as it's just returning the result.